pkg/util/hostname.go

```diff
@@ -228,6 +228,11 @@ func GetHostnameData() (HostnameData, error) {
 	// and the hostname is one of the default ones
 	if getEC2Hostname, found := hostname.ProviderCatalog["ec2"]; found {
 		log.Debug("GetHostname trying EC2 metadata...")
+		if !ecs.IsECSInstance() && ec2.IsWindowsDefaultHostname(hostName) && !config.Datadog.GetBool("ec2_use_windows_prefix_detection") {
+			// REMOVEME: This should be removed when the default `ec2_use_windows_prefix_detection` is set to true
+			log.Warn("You may want to use the EC2 instance-id for the in-app hostname." +
```

Comment:
can we detect whether `ec2_use_windows_prefix_detection` is set in the configuration file? (not sure if `config.Datadog.IsSet` actually works as expected since we set a default on this config key)

If we can't, then there's no way for a user who sees this warning to turn it off. If we can't, and since we're not sure at this point whether we want to change the default of `ec2_use_windows_prefix_detection` in the long run, I think we should lower the level of this log entry to `Info`.

Reply:
2nd question: at this point in the logic, are we sure that the message will be logged _if and only if_ the in-app hostname would be different with `ec2_use_windows_prefix_detection` set to `true`?

(ideally, that's what we should aim for: show the message for Agents that would see their hostname change if `ec2_use_windows_prefix_detection` is set to true, and show it only for these Agents)

Reply:
> can we detect whether ec2_use_windows_prefix_detection is set in the configuration file? (not sure if config.Datadog.IsSet actually works as expected since we set a default on this config key)

I tested on mac and `config.Datadog.IsSet("ec2_use_windows_prefix_detection")` is false even when `ec2_use_windows_prefix_detection` is not set in datadog.yaml. Moreover on Windows (Where this flag should be used) all values are set to default in `datadog.yaml`

> If we can't, then there's no way for a user who sees this warning to turn it off. If we can't, and since we're not sure at this point whether we want to change the default of ec2_use_windows_prefix_detection in the long run, I think we should lower the level of this log entry to Info

Good point. Do we also want to update`hostname_fqdn` warning to `info` https://github.com/DataDog/datadog-agent/blob/98a6c25b581f381d80a90bb408d30dcf8620d015/pkg/util/hostname.go#L232 I am not sure we will change it in the long run and I do not think there is a way to turn off this warning?

Reply:
> I tested on mac and config.Datadog.IsSet("ec2_use_windows_prefix_detection") is false even when ec2_use_windows_prefix_detection is not set in datadog.yaml. Moreover on Windows (Where this flag should be used) all values are set to default in datadog.yaml

Ok, overall what I want is that we don't log the warning for users who have this value set in `datadog.yaml`, since we know that they will not be affected by an upgrade that would change the default value of this option to `true`. So if there's a way to detect this, let's implement it.

> I am not sure we will change it in the long run and I do not think there is a way to turn off this warning?

For `hostname_fqdn`, I'm pretty sure we'll want to change this default value in the long run, because using a non-FQDN hostname leads to host aliasing issues. So I'd keep it as a warning.

Reply:
> I tested on mac and config.Datadog.IsSet("ec2_use_windows_prefix_detection") is false 

I made a typo `config.Datadog.IsSet("ec2_use_windows_prefix_detection")` is `true`

Reply:
ok, then let's use `log.Info` for `ec2_use_windows_prefix_detection` 👍 

What do you think of my "2nd question":
>  at this point in the logic, are we sure that the message will be logged if and only if the in-app hostname would be different with ec2_use_windows_prefix_detection set to true?